src/pages/index.js

```diff
@@ -66,57 +49,201 @@ const cardGallerySelector = ".gallery__grid";
 //access the card template
 const cardSelector = "#card-template";
 
-//function to create card
 let place = "append";
+const cardHolder = [];
+//function to create card
 const createCard = (item) => {
-  const card = new Card(item, cardSelector, handleImageClick);
+  const card = new Card({
+    data: item,
+    cardSelector: cardSelector,
+    handleImageClick: handleImageClick,
+    handleTrashClick: () => {
+      const cardId = card.getId();
+      confirmModal.setAction(() => handleDeleteConfirm(card, cardId));
+      confirmModal.open();
+    },
+    toggleLike: () => {
+      const cardId = card.getId();
+      const isLiked = card.isLiked();
+      //if it already is liked
+      if (isLiked) {
+        newApi
+          .removeLike({ cardId: cardId, isLiked: isLiked })
+          .then((res) => {
+            card.setIsLiked(false);
+            console.log(`disliking: ${cardId}`);
+            console.log(res);
+          })
+          .catch((err) => {
+            console.error("Error. The request has failed: ", err);
+          });
+      }
+      //if it is not liked:
+      else {
+        newApi
+          .addLike({ cardId: cardId, isLiked: isLiked })
+          .then((res) => {
+            card.setIsLiked(true);
+            console.log(`liking: ${cardId}`);
+            console.log(res);
+          })
+          .catch((err) => {
+            console.error("Error. The request has failed: ", err);
+          });
+      }
+    },
+  });
   const cardElement = card.getView();
-
   section.addItem(cardElement, place);
+
+  return { card, cardId: card.getId() };
 };
 const section = new Section(
-  { items: initialCards, renderer: createCard },
+  { items: cardHolder, renderer: createCard },
   cardGallerySelector
 );
-section.renderItems();
 
+const handleDeleteConfirm = (currentCard, cardId) => {
+  newApi
```

Comment:
Needs correction: don't forget to notify the user about loading by changing the button text

Reply:
The TripleTen site says my project has been approved before I fixed these 2 issues, so I cannot send you the pull request link via the TripleTen submission. Please let me know if it is okay to merge. Thank you so much for all your help!

Reply:
> The TripleTen site says my project has been approved before I fixed these 2 issues, so I cannot send you the pull request link via the TripleTen submission. Please let me know if it is okay to merge. Thank you so much for all your help!

Hi! Yes, everything is ok, your project is ready to merge.